protocols/gossipsub/src/behaviour/tests.rs

```diff
@@ -108,32 +107,62 @@ where
 
         (gs, peers, topic_hashes)
     }
-}
 
-impl<D, F> InjectNodesBuilder<D, F>
-where
-    D: DataTransform + Default + Clone + Send + 'static,
-    F: TopicSubscriptionFilter + Clone + Default + Send + 'static,
-{
-    pub fn create_network(&self) -> (Gossipsub<D, F>, Vec<PeerId>, Vec<TopicHash>) {
-        self.build().unwrap().create_network()
+    fn peer_no(mut self, peer_no: usize) -> Self {
+        self.peer_no = peer_no;
+        self
+    }
+
+    fn topics(mut self, topics: Vec<String>) -> Self {
+        self.topics = topics;
+        self
+    }
+
+    fn to_subscribe(mut self, to_subscribe: bool) -> Self {
+        self.to_subscribe = to_subscribe;
+        self
+    }
+
+    fn gs_config(mut self, gs_config: GossipsubConfig) -> Self {
+        self.gs_config = gs_config;
+        self
+    }
+
+    fn explicit(mut self, explicit: usize) -> Self {
+        self.explicit = explicit;
+        self
+    }
+
+    fn outbound(mut self, outbound: usize) -> Self {
+        self.outbound = outbound;
+        self
+    }
+
+    fn scoring(mut self, scoring: Option<(PeerScoreParams, PeerScoreThresholds)>) -> Self {
+        self.scoring = scoring;
+        self
+    }
+
+    fn subscription_filter(mut self, subscription_filter: F) -> Self {
+        self.subscription_filter = subscription_filter;
+        self
     }
 }
 
-fn inject_nodes<D, F>() -> InjectNodesBuilder<D, F>
+// helper functions for testing
+
+fn inject_nodes<D, F>() -> InjectNodes<D, F>
 where
     D: DataTransform + Default + Clone + Send + 'static,
     F: TopicSubscriptionFilter + Clone + Default + Send + 'static,
 {
-    InjectNodesBuilder::default()
+    InjectNodes::default()
 }
 
-fn inject_nodes1() -> InjectNodesBuilder<IdentityTransform, AllowAllSubscriptionFilter> {
-    inject_nodes()
+fn inject_nodes1() -> InjectNodes<IdentityTransform, AllowAllSubscriptionFilter> {
+    InjectNodes::<IdentityTransform, AllowAllSubscriptionFilter>::default()
 }
 
-// helper functions for testing
```

Comment:
Nit: Can we retain the position of this comment?

Reply:
:heavy_check_mark: